x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java

```diff
@@ -143,11 +165,11 @@ protected static class TransportRequest extends AbstractLookupService.TransportR
             Page inputPage,
             Page toRelease,
             List<NamedExpression> extractFields,
-            String matchField,
+            List<LookupFromIndexOperator.MatchConfig> matchFields,
```

Comment:
The channel information doesn't matter here and in the `Request`, right? Let's add comments; this is probably fine, but it could matter for e.g. future optimizations that we're sending redundant data over the wire. (We had performance problems from redundant serialized data in the past.)